src/Symfony/Component/JsonPath/Functions/JsonPathFunctionsProvider.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonPath\Functions;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class JsonPathFunctionsProvider implements JsonPathFunctionsProviderInterface
+{
+    /**
+     * @var array<string, JsonPathFunctionInterface>
+     */
+    private array $functions = [];
+
+    /**
+     * @param iterable<JsonPathFunctionInterface> $functions
+     */
+    public function __construct(iterable $functions = [])
+    {
+        foreach ($functions as $function) {
+            $this->addFunction($function);
```

Comment:
instead of injecting an iterable that is used in the constructor (breaking any lazy-loading of the iterator argument provided by the DI component), I suggest injecting a service locator instead (which would require knowing the name statically, which could be done by always specifying the name in the attribute instead of the `getName` method in the instance)